x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/GetTrainedModelsStatsAction.java

```diff
@@ -0,0 +1,190 @@
+import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.ingest.IngestStats;
+import org.elasticsearch.xpack.core.action.AbstractGetResourcesRequest;
+import org.elasticsearch.xpack.core.action.AbstractGetResourcesResponse;
+import org.elasticsearch.xpack.core.action.util.QueryPage;
+import org.elasticsearch.xpack.core.ml.inference.TrainedModelConfig;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class GetTrainedModelsStatsAction extends ActionType<GetTrainedModelsStatsAction.Response> {
+
+    public static final GetTrainedModelsStatsAction INSTANCE = new GetTrainedModelsStatsAction();
+    public static final String NAME = "cluster:monitor/xpack/ml/inference/stats/get";
+
+    public static final ParseField MODEL_ID = new ParseField("model_id");
+    public static final ParseField PIPELINE_COUNT = new ParseField("pipeline_count");
+
+    private GetTrainedModelsStatsAction() {
+        super(NAME, GetTrainedModelsStatsAction.Response::new);
+    }
+
+    public static class Request extends AbstractGetResourcesRequest {
+
+        public static final ParseField ALLOW_NO_MATCH = new ParseField("allow_no_match");
+
+        public Request() {
+            setAllowNoResources(true);
+        }
+
+        public Request(String id) {
+            setResourceId(id);
+            setAllowNoResources(true);
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        @Override
+        public String getResourceIdField() {
+            return TrainedModelConfig.MODEL_ID.getPreferredName();
+        }
+
+    }
+
+    public static class RequestBuilder extends ActionRequestBuilder<Request, Response> {
+
+        public RequestBuilder(ElasticsearchClient client, GetTrainedModelsStatsAction action) {
+            super(client, action, new Request());
+        }
+    }
+
+    public static class Response extends AbstractGetResourcesResponse<Response.TrainedModelStats> {
+
+        public static class TrainedModelStats implements ToXContentObject, Writeable {
+            private final String modelId;
+            private final IngestStats ingestStats;
+            private final int pipelineCount;
+
+            private static final IngestStats EMPTY_INGEST_STATS = new IngestStats(new IngestStats.Stats(0, 0, 0, 0),
+                Collections.emptyList(),
+                Collections.emptyMap());
+
+            public TrainedModelStats(String modelId, IngestStats ingestStats, int pipelineCount) {
```

Comment:
`pipelineCount` is `int` because it is the size of the pipeline stats array. Java returns an `int` from a `size()` call.

As for `modelCount` it is the total number of models matched by the provided ids. If `_all` is provided, then the total number of models is returned. 

Additionally, the underlying `QueryPage` object assumes that the `count` value is `long`.